### ISSUE TITLE
useUpdateLoading is missing in v4

### ISSUE THREAD

**Issue by 7system7** [marmelab/react-admin#7554]
**What you were expecting:**
```javascript
import { useUpdateLoading } from "react-admin";
...
const {startLoading, stopLoading} = useUpdateLoading();
```

**Btw, in my opinion, it is an important feature, so I hope it hasn't been removed from the source...**

**What happened instead:**
"Cannot resolve symbol useUpdateLoading". So, it seems, the feature completely missing from the source, however, in the documentation, [it is still exists](https://marmelab.com/react-admin/Reference.html).

**Steps to reproduce:**
1. Install the latest (v4) version of `react-admin`
2. import the `useUpdateLoading` hook
3. see the error msg

**Other information:**

**Environment**

* React-admin version: 4.0.0
* Last version that did not exhibit the issue (if applicable): 3.19.11
* React version: 18.0.0
* Browser: Firefox 99.0.1
* Stack trace (in case of a JS error):
```
export 'useUpdateLoading' (imported as 'useUpdateLoading') was not found in 'react-admin' (possible exports: AUTH_CHECK, ...
```

**Comment by fzaninotto** [comment]
Hi, and thanks for your feedback.

In v4, the loading state is now handled by react-query, so `useUpdateLoading` is no longer necessary. We've removed this hook from the core, and we should remove it from the documentation, too (I'm marking this issue as a documentation bug). 

If you want to trigger the global spinner, all you have to do is wrap your async call in a call to react-query's `useQuery`.

**Comment by fzaninotto** [comment]
Sorry, I mistakenly pushed the fix to `master`. It will be released in 4.0.3.
